common/dconfig/configmgr.go

```diff
@@ -318,14 +325,37 @@ func (cfgMgr *CassandraConfigManager) mkConfigTreeForSvc(cfgType interface{}, de
 	return result
 }
 
+// mkKVTreeForSvc builds a hierarchical tree of key values
+// for a speicific service. The tree hierarchy is from
+// svc -> version -> sku -> host. The key values at each
+// level serve as the default for its children nodes.
+//
+// The actual construction of the tree is a 3 step process:
+// (1) Take every config item of form svc.version.sku.host.key=value and add it to tree
+//        (a) If the version & sku are wildcard (i.e. *), then the config item
+//            MUST serve as a default for every version/ every sku, so keep track of
+//            these separately
+//        (b) If the version is a wildcard but sku is not, then this config item
+//			  MUST serve as a default for every version.sku, so trace this separately
+//        (c) If the version is not wildcard, but sku is wildcard, then the
+//            config item MUST serve as a default for every sku for that version
+//            so keep track of these separately
+// (2) Add the items tracked in 1(b) to the tree (for every sku)
+// (3) Add the items tracked in 1(a) for every version
+//
+// Its important to apply the config items in the order (1).(2).(3) to honor
+// the hierarchial structure
 func (cfgMgr *CassandraConfigManager) mkKVTreeForSvc(service string, items []*m.ServiceConfigItem) *kvTreeNode {
 
 	root := newKVTreeNode()
 
-	versions := make(map[string]struct{})
-	wildcardVersionItems := make([]*m.ServiceConfigItem, 0, len(items))
+	allSkus := make(map[string]struct{})
+	allVersions := make(map[string]struct{})
+	wildcardVersionItems := make([]*m.ServiceConfigItem, 0, len(items)) // items with version as wildcard
+	wildcardVersionSkuItems := make([]*m.ServiceConfigItem, 0, len(items)) // items with version as wildcard and sku not a wildcard
+	versionToWildcardSkuItems := make(map[string][]*m.ServiceConfigItem, len(items)) // items with sku as wildcard
```

Comment:
cosmetic: my editor/lint seems to align up the comments on these three lines to the same column.